docs/src/main/docs/guides/07_openapi.adoc

```diff
@@ -0,0 +1,240 @@
+
+The next sections show you how to make these changes. Helidon also includes a 
+link:{helidon-mp-openapi-example}[complete example maven project] showing
+Helidon MP OpenAPI support.
+
+=== Annotating endpoints
+The added OpenAPI-related annotations allow the system to build the OpenAPI document 
+describing your endpoints. 
+
+Here is one of the endpoints from the example mentioned earlier:
+
+[source,java]
+----
+@GET
+@Operation(summary = "Returns a generic greeting", // <1>
+        description = "Greets the user generically")
+@APIResponse(description = "Simple JSON containing the greeting", // <2>
+        content = @Content(mediaType = "application/json",
+                           schema = @Schema(implementation = GreetingMessage.class)))
+@Produces(MediaType.APPLICATION_JSON)
+public JsonObject getDefaultMessage() {...}
+----
+<1> `@Operation` gives information about this endpoint.
+<2> `@APIResponse` describes the HTTP response and declares its media type and contents.
+
+You can also define any request parameters the endpoint expects, although this
+endpoint uses none.
+
+This guide shows only a few annotations for illustration. The 
+link:{helidon-mp-openapi-example}[Helidon MP OpenAPI example] illustrates more, 
+and the link:{mp-openapi-spec}[MicroProfile OpenAPI spec] describes them all.
+
+=== Updating your `pom.xml`
+==== Building the Jandex index
+A Jandex index stores information about the classes and methods in your app and
+what annotations they have. It allows CDI to process annotations faster during your 
+application's start-up.
+
+Add the link:{jandex-plugin-doc}[Jandex maven plug-in] to the `<build><plugins>` 
+section of your `pom.xml`:
+
+[source,xml]
+----
+<plugin>
+    <groupId>org.jboss.jandex</groupId>
+    <artifactId>jandex-maven-plugin</artifactId>
+    <executions>
+      <execution>
+        <id>make-index</id>
+        <goals>
+          <goal>jandex</goal>
+        </goals>
+      </execution>
+    </executions>
+</plugin>
+----
+You will find `META-INF/jandex.idx` in your JAR.
+
+[NOTE]
+====
+If you _do not_ modify your build to create 
+the index then the Helidon MP OpenAPI runtime automatically creates one in memory during 
+app start-up. This slows down your app start-up and, depending on how CDI is 
+configured, might inadvertently miss information. 
+
+We _strongly recommend_ building the index into your app.
+====
+
+==== Adding Helidon MP OpenAPI support to your app
+Add these two sections to the `<dependencies>` portion of your `pom.xml`:
+
+[source,xml,subs="attributes+"]
+----
+<dependency> <!--1-->
+    <groupId>org.eclipse.microprofile.openapi</groupId>
+    <artifactId>microprofile-openapi-api</artifactId>
+    <version>{microprofile-openapi-version}</version>
+</dependency>
+
+<dependency> <!--2-->
+    <groupId>io.helidon.microprofile.openapi</groupId>
+    <artifactId>helidon-microprofile-openapi</artifactId>
+    <version>{helidon-version}</version>
```

Comment:
Typo; need a `$` before the `{`

Reply:
See earlier note about omitting the `$`.